### PR TITLE
[O365 File Management] Add Auth code authentication support

### PR DESCRIPTION
**Pull request by michal-dagan**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-9273)

## Description
https://learn.microsoft.com/en-us/answers/questions/1155463/graph-api-delegated-permissions-for-sites-selected

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py [hunk 1]
```diff
@@ -511,19 +526,20 @@ def create_new_folder(self, object_type: str, object_type_id: str, parent_id: st
...
-    try:
-        client.ms_client.http_request(
-            url_suffix="sites",
-            timeout=7,
-            method="GET")
+    response = 'ok' if demisto.command() == 'test-module' else '```✅ Success!```'
+    if demisto.params().get('self_deployed', False) and demisto.command() == 'test-module':
+        if (client.ms_client.grant_type == AUTHORIZATION_CODE
+                or demisto.params().get('redirect_uri') or demisto.params().get('auth_code_creds', {}).get('password', '')):
+            raise DemistoException("The *Test* button is not available for the `self-deployed - Authorization Code Flow`.\n "
```
Comment (michal-dagan) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py:537]: @ShirleyDenkberg please review the DemistoException also

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.yml [hunk 2]
```diff
@@ -737,6 +749,13 @@ script:
       required: true
     name: msgraph-delete-site-permissions
     description: Deletes an app permission from a site.
+  - description: Tests connectivity to Microsoft.
+    execution: false
+    name: msgraph-files-auth-test
+    arguments: []
+  - description: Generate the login url used for Authorization code flow.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.yml:756]:
```suggestion
  - description: Generate the login URL used for Authorization code flow.
```

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_description.md [hunk 3]
```diff
@@ -11,8 +12,9 @@ There are two application authentication methods available:
 
 Depending on the authentication method that you use, the integration parameters might change.
 
-To allow access to O365 File Management (Onedrive/Sharepoint/Teams), an administrator has to approve the Cortex XSOAR app using the [admin consent flow](https://oproxy.demisto.ninja/ms-graph-files).
-After authorizing the Cortex XSOAR app, you receive an ID, Token, and Key, all of which need to be entered to the corresponding fields when configuring the integration instance.
+
+To use the **Cortex XSOAR application** and allow us access to O365 File Management (Onedrive/Sharepoint/Teams), an administrator has to approve our app using an admin consent flow by clicking this **[link](https://oproxy.demisto.ninja/ms-graph-files)**.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_description.md:16]: Who is "us"?

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/README.md [hunk 4]
```diff
@@ -6,10 +6,14 @@ For more details about the authentication used in this integration, see <a href=
 
 ### Required Permissions
 The required permission depends on whether you want to access all sites (Sites.ReadWrite.All) or specific sites (Site.Selected):
-- `Sites.ReadWrite.All - Application`: Provides read and write access to all sites.
+- `Sites.ReadWrite.All`: Provides read and write access to all sites.\
+`Client Credentials Flow` - Application permission.\
+`Authorization Code Flow` - Delegated permission.\
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/README.md:11]:
```suggestion
- `Sites.ReadWrite.All`: Provides read and write access to all sites.
- `Client Credentials Flow` - Application permission.
- `Authorization Code Flow` - Delegated permission.
```

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/README.md [hunk 5]
```diff
@@ -956,3 +979,53 @@ There is no context output for this command.
...
+
+There are no input arguments for this command.
+
+#### Context Output
+
+There is no context output for this command.
+### msgraph-files-generate-login-url
+
+***
+Generate the login url used for Authorization code flow.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/README.md:1002]:
```suggestion
Generate the login URL used for Authorization code flow.
```

File: Packs/MicrosoftGraphFiles/ReleaseNotes/1_1_21.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### O365 File Management (Onedrive/Sharepoint/Teams)
+
+- Added support for *Authorization Code Flow* which enables operations with delegated permissions.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphFiles/ReleaseNotes/1_1_21.md:6]:
```suggestion
Added support for *Authorization Code Flow* which enables operations with delegated permissions.
```

**Comment by ShirleyDenkberg** [comment]
@DeanArbel Doc review completed.

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_description.md (hunk 3 above)
Comment (michal-dagan) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_description.md:16]:
```suggestion
To use the **Cortex XSOAR application** and allow Cortex XSOAR access to O365 File Management (Onedrive/Sharepoint/Teams), an administrator has to approve our app using an admin consent flow by clicking this **[link](https://oproxy.demisto.ninja/ms-graph-files)**.
```

**Pull request merged**